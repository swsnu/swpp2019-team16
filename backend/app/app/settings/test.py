from .base import *
import environ
from redis import ConnectionPool

env = environ.Env()
env.read_env('env/.env.development')

DEBUG = True

# Database
# https://docs.djangoproject.com/en/2.1/ref/settings/#databases

DATABASES = {
    'default': {
        'ENGINE': 'django.db.backends.sqlite3',
        'NAME': os.path.join(BASE_DIR, 'db.sqlite3'),
    }
}

REDIS_CONNECTION_POOL = ConnectionPool(host=env('REDIS_HOST'), port=6379, db=0)
