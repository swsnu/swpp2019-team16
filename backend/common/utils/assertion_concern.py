class AssertionConcern:

    def assert_not_none(self, val, msg):
        if val is None:
            raise TypeError(msg)
