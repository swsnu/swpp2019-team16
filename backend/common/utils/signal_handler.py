import asyncio
import functools
import signal


def register_signal_handler(loop, shutdown):
    for signame in ('SIGINT', 'SIGTERM'):
        loop.add_signal_handler(
            getattr(signal, signame),
            functools.partial(asyncio.ensure_future,
                              shutdown(signame, loop)))


async def shutdown_process(self, sig, loop):
    print('caught {0}'.format(sig))
    tasks = [task for task in asyncio.Task.all_tasks() if task is not
             asyncio.tasks.Task.current_task()]

    list(map(lambda task: task.cancel(), tasks))
    results = await asyncio.gather(*tasks, return_exceptions=True)

    print('finished awaiting cancelled tasks, results: {0}'
          .format(results))
    loop.stop()
