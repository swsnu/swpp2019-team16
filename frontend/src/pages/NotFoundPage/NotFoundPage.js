import React from 'react';
import Main from '../_layouts/Main';
import NotFound from '../../components/NotFound';

function NotFoundPage() {
  return (
    <Main>
      <NotFound />
    </Main>
  );
}
export default NotFoundPage;
