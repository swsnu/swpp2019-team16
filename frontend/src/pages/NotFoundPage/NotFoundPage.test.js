import React from 'react';
import NotFoundPage from './NotFoundPage';
import { MemoryRouter, Route } from 'react-router-dom';
import { renderWithRedux } from 'test/utils';

describe('<NotFoundPage />', () => {
  const initialState = {
    user: {
      user: null,
    },
    auth: {
      auth: null,
    },
  };
  it('SHOULD match with snapshot', async () => {
    const { container } = renderWithRedux(
      <MemoryRouter initialEntries={['']}>
        <Route component={NotFoundPage} />
      </MemoryRouter>,
      initialState,
    );
    expect(container).toMatchSnapshot();
  });
});
