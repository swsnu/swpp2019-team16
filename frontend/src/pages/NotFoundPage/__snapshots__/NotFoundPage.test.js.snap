// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<NotFoundPage /> SHOULD match with snapshot 1`] = `
<div>
  <div
    class="sc-bdVaJa bPGLpq"
  >
    <div
      class="makeStyles-root-1"
    >
      <div
        class="MuiToolbar-root MuiToolbar-regular MuiToolbar-gutters"
      >
        <h4
          class="MuiTypography-root makeStyles-title-3 MuiTypography-h4"
        >
          Ya-Ta!
        </h4>
        <div
          class="makeStyles-button-4"
        >
          LOGIN
        </div>
      </div>
    </div>
    <div
      class="makeStyles-root-39"
    >
      <div
        class="makeStyles-whiteBox-41"
      >
        <div
          class="makeStyles-container-40"
        >
          <h1
            class="MuiTypography-root MuiTypography-h1"
          >
            404
          </h1>
        </div>
        <div
          class="makeStyles-container-40"
        >
          <h1
            class="MuiTypography-root MuiTypography-h1"
          >
            Page Not Found
          </h1>
        </div>
        <div
          class="makeStyles-container-40"
        >
          <h2
            class="MuiTypography-root MuiTypography-h2"
          >
            This is not the page you are looking for
          </h2>
        </div>
        <div
          class="makeStyles-container-40"
        >
          <div
            class="makeStyles-button-42"
            fullwidth="false"
            variant="contained"
          >
            Go Home
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
